engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schema/DatumReaderWriterMixin.scala

```diff
@@ -21,19 +21,22 @@ trait DatumReaderWriterMixin {
     */
   protected val primitives: mutable.Map[String, Schema] = AvroSchemaUtils.getPrimitiveSchemas.asScala
 
-
-  def createDatumWriter(record: Any, schema: Schema, useSchemaReflection: Boolean): GenericDatumWriter[Any] = record match {
-    case _: SpecificRecord => new SpecificDatumWriter[Any](schema, AvroUtils.specificData)
-    case _ if useSchemaReflection => new ReflectDatumWriter[Any](schema, AvroUtils.reflectData)
-    case _ => new GenericDatumWriter[Any](schema, AvroUtils.genericData)
+  def createDatumWriter(schema: Schema, useSchemaReflection: Boolean, useSpecificAvroReader: Boolean): GenericDatumWriter[Any] = {
+    if (useSpecificAvroReader) {
+      new SpecificDatumWriter[Any](schema, AvroUtils.specificData)
+    } else if (useSchemaReflection) {
+      new ReflectDatumWriter[Any](schema, AvroUtils.reflectData)
+    } else {
+      new GenericDatumWriter[Any](schema, AvroUtils.genericData)
+    }
   }
 
-  def createDatumReader(writerSchema: Schema, readerSchema: Schema, useSchemaReflection: Boolean, useSpecificAvroReader: Boolean): DatumReader[AnyRef] = {
+  def createDatumReader(writerSchema: Schema, readerSchema: Schema, useSchemaReflection: Boolean, useSpecificAvroWriter: Boolean): DatumReader[AnyRef] = {
```

Comment:
Why is it reader here?

Reply:
Exaclty, why you want to specify that should be used specificAvroWriter in method returning reader? :)

Reply:
Change reverted. My point was to use `DatumReaderWriterMixin` in `ConfluentAvroMessageReader` and `ConfluentAvroMessageFormatter`. I've found a better solution.